### PR TITLE
Merge feature branch

### PR DESCRIPTION
**Pull request by davidauza-engineer**
Please let us know your comments about our work.

### CODE REVIEW

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require './game.rb'
+
+describe Game do
+  subject(:game) { Game.new }
```
Comment (stratospherique) [inline spec/tic_tac_toe_spec.rb]: - This line is causing interactions with the user when we run **rspec**
- Use let to define test cases variables

File: spec/tic_tac_toe_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  describe '#over?' do
+    it 'returns true when a win condition is met' do
+      (1..3).each do |i|
+        game.player_one.add_cell(i)
+      end
+      expect(game.over?).to be true
+    end
+  end
+end
```
Comment (stratospherique) [inline spec/tic_tac_toe_spec.rb]: - You should be testing all the winning conditions :nerd_face:

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Good start team :clap:. However, the specified tests doesn't meet with the project's requirements. check below :arrow_down: 

* [x] All the public methods of the classes implemented in project OOP are tested
   - not all public methods are tested in your specs. :no_entry_sign: 
* [x] The specs should cover positive and negative scenarios Example: there is a spec for correct usage of method \[e.g. “it “creates object if all required attributes are passed”\] and also there is spec for wrong usage of method \[e.g. “it “does not create object if required attributes are missing”\].

* [x] Specs for each method should be wrapped in one `describe` block. As in example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples .
* [x] You should have one `it` block for each expected behaviour.

* [x] to define a variable for the test cases, students should [Use `let` instead](https://github.com/rubocop-hq/rspec-style-guide#let-blocks).
 
###### PS: check the code comment below :arrow_down: 

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

happy coding :v: 
**Ahmed Mahfoudh**

File: spec/tic_tac_toe_spec.rb (hunk 1 above)
Comment (davidauza-engineer) [reply spec/tic_tac_toe_spec.rb]: Done, thank you so much :)

File: spec/tic_tac_toe_spec.rb (hunk 2 above)
Comment (davidauza-engineer) [reply spec/tic_tac_toe_spec.rb]: Added one case for every winning condition as requested 👍

**Comment by davidauza-engineer** [comment]
We made all the requested changes, please let us know if everything is OK now. Thanks in advance :)

**Review by Ispirett — CHANGES_REQUESTED** [review]
## Status :changes required
Great job writing the tests @davidauza-engineer @falguniislam and implementing changes :+1:  :100: 

Unfortunately, im getting a lot broken test and  other bugs :bug: 

- [x] I should be  able to run rspec from the main folder. When I do the tictactoe game runs
- [x] You should prefix describe classes with like `**RSpec.describe player do**`
- [x] errors bellow when running your tests
rspec ./spec/tic_tac_toe_spec.rb:11 # Board#draw_board draw board appropriately
rspec ./spec/tic_tac_toe_spec.rb:31 # Game#play_turn sets a turn adequately returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:37 # Game#populate_display populates the display appropriately returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:43 # Game switch_player successfully switch the player by returning 0 for success
rspec ./spec/tic_tac_toe_spec.rb:49 # Game over? successfully finish the game with the first winning condition
rspec ./spec/tic_tac_toe_spec.rb:53 # Game over? successfully finish the game with the second winning condition
rspec ./spec/tic_tac_toe_spec.rb:57 # Game over? successfully finish the game with the third winning condition
rspec ./spec/tic_tac_toe_spec.rb:61 # Game over? successfully finish the game with the fourth winning condition
rspec ./spec/tic_tac_toe_spec.rb:65 # Game over? successfully finish the game with the fifth winning condition
rspec ./spec/tic_tac_toe_spec.rb:69 # Game over? successfully finish the game with the sixth winning condition
rspec ./spec/tic_tac_toe_spec.rb:73 # Game over? successfully finish the game with the seventh winning condition
rspec ./spec/tic_tac_toe_spec.rb:77 # Game over? successfully finish the game with the eighth winning condition
rspec ./spec/tic_tac_toe_spec.rb:81 # Game over? continue the game if the given array is not one from the winning options

**Comment by davidauza-engineer** [comment]
We modified the necessary files to successfully run rspec from the main folder, prefixed the describe classes as requested and fixed all the errors:

Screen Shot 2019-10-25 at 12 16 01 PM

We'll be waiting for your feedback :)

**Review by wathigo — APPROVED** [review]
### Hi @davidauza-engineer  and @falguniislam  :wave:  
Great work implementing the requested changes.

**You may move on :next_track_button:**  

_Have a great weekend :smile: 
Reviewed by: **Simon Wathigo**_

**Comment by davidauza-engineer** [comment]
Hi @stratospherique , @Ispirett  and @wathigo. Thank you so much for your comments and those great reviews 👍

**Pull request merged**